docs/software-architecture-and-design.md

```diff
@@ -196,7 +197,7 @@ In our system, we need to be able to do multiple things (receive camera data, ru
 **WARNING:** If a class extends multiple `ThreadedObserver`s (for example, [AI](#ai) could extend `ThreadedObserver<World>` and `ThreadedObserver<RobotStatus>`), then there will be two threads running, one for each observer. We **do not check** for data race conditions between observers, so it's entirely possible that one `ThreadedObserver` thread could read/write from data at the same time as the other `ThreadedObserver` is reading/writing the same data. Please make sure any data read/written to/from multiple `ThreadedObserver`s is thread-safe.
 
 ### Example
-One example of this is the `Backend`, which extends `Subject<World>` and the [AI](#ai), which extends `ThreadedObserver<World>`. The backend runs in one thread and sends data to the [AI](#ai), which receives and processes it another thread.
+One example of this is [SensorFusion](#sensor-fusion), which extends `Subject<World>` and the [AI](#ai), which extends `ThreadedObserver<World>`. [SensorFusion](#sensor-fusion) runs in one thread and sends data to the [AI](#ai), which receives and processes it another thread.
 
 ## C++ Templating
 While debatably not a design pattern depending on who you ask, templating in C++ is a powerful tool that is very useful to understand. [https://www.geeksforgeeks.org/templates-cpp/] gives a great explanantion and example.
```

Comment:
In line 243-244, "which is uses" -> "which it uses"

(for some reason I cannot add a comment to that line)

Reply:
In line 251-252, "Asymetric Coroutines" -> "Asymmetric Coroutines"

Reply:
ah that's because it's not in the diff for this PR. I've fixed those typos though